src/main/java/alpha/Parser.java

```diff
@@ -0,0 +1,238 @@
+                return uncheck(input);
+            } else if (splitInput[0].equals(DEADLINE)) {
+                return addDeadline(input);
+            } else if (splitInput[0].equals(EVENT)) {
+                return addEvent(input);
+            } else if (splitInput[0].equals(TODO)) {
+                return addToDo(input);
+            } else if (input.equals(TODO)) {
+                return delete(input);
+            } else if (input.equals(END)) {
+                return new ExitCommand(taskList, fileHandler, ui);
+            } else if (splitInput[0].equals(DELETE)) {
+                return delete(input);
+            } else if (splitInput[0].equals(FIND)) {
+                return find(input);
+            } else {
+                throw new InvalidInputException("Invalid Input!");
+            }
+        } catch (MissingIndexException e1) {
+            return new InvalidCommand(taskList, fileHandler, ui, e1.getMessage() +
+                    " Please enter the index of the number you would like to mark.");
+        } catch (InvalidIndexException e2) {
+            return new InvalidCommand(taskList, fileHandler, ui, e2.getMessage() +
+                    " Please enter a valid index. To check all valid indices, " +
+                    "type \"list\" and press ENTER");
+        } catch (InvalidInputException e3) {
+            return new InvalidCommand(taskList, fileHandler, ui, e3.getMessage()
+                    + " Please input something meaningful.");
+        } catch (NumberFormatException e4) {
+            return new InvalidCommand(taskList, fileHandler, ui, "Please enter a number.");
+        } catch (InvalidFormatException | MissingInfoException e5) {
+            if (e5.getTask() == TaskException.TaskType.TODO) {
+                return new InvalidCommand(taskList, fileHandler, ui, e5.getMessage()
+                        + " Please enter a todo in the " +
+                        "format \"todo YOUR_DESCRIPTION\"");
+            } else if (e5.getTask() == TaskException.TaskType.DEADLINE) {
+                return new InvalidCommand(taskList, fileHandler, ui, e5.getMessage()
+                        + " Please enter a deadline in the format " +
+                        "\"deadline YOUR_DESCRIPTION /by YOUR_TIME\" ");
+            } else if (e5.getTask() == TaskException.TaskType.EVENT) {
+                return new InvalidCommand(taskList, fileHandler, ui,
+                        " Please enter an event in the format " +
+                        "\"event YOUR_DESCRIPTION " +
+                        "/from START_TIME /to END_TIME\" ");
+            }
+            return new InvalidCommand(taskList, fileHandler, ui, "");
+        } catch (DateTimeParseException e6) {
+            return new InvalidCommand(taskList, fileHandler, ui,
+                    e6.getMessage() +"The date is in an invalid format! " +
+                            "Enter the date in the format YYYY-MM-DD");
+        }
+    }
```

Comment:
This method is quite long, maybe instead of using if else, you can use switch case to reduce space. Alternatively can remove some of the new lines / enters in the lines so it would not consume too many lines